caddytls/config.go

```diff
@@ -281,19 +279,20 @@ func (c *Config) StorageFor(caURL string) (Storage, error) {
 
 	// Create the storage based on the URL
 	var s Storage
-	if c.StorageCreator != nil {
-		s, err = c.StorageCreator(u)
-		if err != nil {
-			return nil, fmt.Errorf("%s: unable to create custom storage: %v", caURL, err)
-		}
+	if c.StorageCreator == "" {
+		c.StorageCreator = "file"
 	}
-	if s == nil {
-		// We trust that this does not return a nil s when there's a nil err
-		s, err = FileStorageCreator(u)
-		if err != nil {
-			return nil, fmt.Errorf("%s: unable to create file storage: %v", caURL, err)
-		}
+
+	creator, ok := storageProviders[c.StorageCreator]
+	if !ok {
+		return nil, fmt.Errorf("%s: Unknown storage: %v", caURL, err)
```

Comment:
This err refers to url.Parse from above. Print the storage provider name instead...